libsbc: update header file comments as doxygen style

MarkWangChinese opened the pull request:
update sbc.h comments as doxygen style.

MarkWangChinese committed:
libsbc: update header file comments as doxygen style

update sbc.h comments as doxygen style.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese commented on include/zephyr/libsbc/sbc.h:
```diff
@@ -15,99 +18,119 @@
 	SBC_ENC_PARAMS sbc_encoder_params;
 };
 
+/** @brief encoder initialization parameters */
```
All are updated, Thanks.

kartben commented on include/zephyr/libsbc/sbc.h:
```diff
@@ -15,99 +18,119 @@
 	SBC_ALLOC_MTHD_SNR,
 };
 
 #if defined(CONFIG_LIBSBC_ENCODER)
```
unless I am missing something, please remove all these `#if defined(CONFIG_LIBSBC_ENCODER)` from this file (right now, among other things, this is hiding the headers to Doxygen - see https://builds.zephyrproject.io/zephyr/pr/95137/api-coverage/zephyr/libsbc/sbc.h.gcov.html)

kartben commented:
Also, as mentioned before, is the top-level folder really the best home for such a specific module header? Couldn't it live somewhere under bluetooth?

MarkWangChinese committed:
libsbc: fix hiding headers to doxygen

remove CONFIG_LIBSBC_ENCODER and CONFIG_LIBSBC_DECODER from sbc.h, then
include the required heders files path defaultly to fix building errors.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese commented on include/zephyr/libsbc/sbc.h:
```diff
@@ -15,99 +18,119 @@
 	SBC_CH_MODE_MONO,
 	SBC_CH_MODE_DUAL_CHANNEL,
 	SBC_CH_MODE_STEREO,
 	SBC_CH_MODE_JOINT_STEREO,
```
Done

MarkWangChinese commented:
> Also, as mentioned before, is the top-level folder really the best home for such a specific module header? Couldn't it live somewhere under bluetooth?

Thanks for the suggestion. The sbc encoder/decoder can be used by any modules, so I put it in the common place before. Since it is only used by bluetooth and I don't know any other modules will use it now, I think they (sbc.h and sbc.c) can be moved to bluetooth.

kartben reviewed (commented):
Almost there, thanks very much!

kartben reviewed (approved):
Many thanks!
